### PR TITLE
Feature/palette selector

### PR DESCRIPTION
**Pull request by mbenfowler**
## What I did:
Added dark and light mode!

Remember to `npm install` because I added a package for the light/dark toggle component

## Testing Notes/Extras: 

1. Ensure all user flows are functioning as expected
2. All e2e tests pass
3. Toggle correctly toggles between light and dark mode.  It's position looks good no matter what size screen you're on.
4. Observe where all the "light" mode colors are being consumed. Do these colors work? Should any other colors be used instead?

Known issues:
1. With variables I am unable to affect the `background-color` we have set on our `html` tag in App.css. It doesn't look terrible, but it is somewhat noticeable especially on light mode.
2. ~~Journal icons are white which makes them invisible in our light mode.  Will need to add more custom icons to fix.~~ I used filter to solve for this

### CODE REVIEW

File: src/components/App/App.css [hunk 1]
```diff
@@ -5,6 +5,7 @@
 }
 
 html {
+  height: 100%;
   background-color: #191725;
   width: 100%;
```
Comment (jalbe0076) [inline src/components/App/App.css:10]: To update the html background colour save the property to a variable and add the colours in the :root
Then just toggle it with a useEffect that re-renders when the theme state changes

:root {
  --background-dark: #191725;
  --background-light: white;
}

html {
  height: 100%;
  background-color: var(--background);
  width: 100%;
}

File: src/components/App/App.js
Comment (jalbe0076) [inline src/components/App/App.js:1]: Add useEffect to update the background colour

  useEffect(() => {
    const root = document.documentElement;
    root.style.setProperty('--background', isDark ? 'var(--background-dark)' : 'var(--background-light)');
  }, [isDark]);

**Review by jalbe0076 — CHANGES_REQUESTED** [review]
I gave you some feedback for how to solve the HTML colour.

File: src/components/App/App.js
Comment (mbenfowler) [reply src/components/App/App.js:1]: oooh yah it was `documentElement` i think I was missing, hadn't seen that one yet

**Review by jalbe0076 — APPROVED** [review]
Looking good!

**Pull request merged**